tools/src/prebuilds/ExpoModulesCorePrebuilder.ts

```diff
@@ -0,0 +1,175 @@
+  await createGeneratedHeaderAsync(workingDir);
+
+  const result = await XcodeGen.generateXcodeProjectAsync(workingDir, spec);
+
+  logger.log('   Installing Pods');
+  await createPodfileAsync(workingDir);
+  await podInstallAsync(workingDir);
+
+  return result;
+}
+
+export async function buildFrameworkAsync(
+  xcodeProject: XcodeProject,
+  target: string,
+  flavor: Flavor,
+  options?: XcodebuildSettings
+): Promise<Framework> {
+  await xcodeProject.xcodebuildAsync(
+    [
+      'build',
+      '-workspace',
+      `${xcodeProject.name}.xcworkspace`,
+      '-scheme',
+      `${target}_iOS`,
+      '-configuration',
+      flavor.configuration,
+      '-sdk',
+      flavor.sdk,
+      ...spreadArgs('-arch', flavor.archs),
+      '-derivedDataPath',
+      SHARED_DERIVED_DATA_DIR,
+    ],
+    options
+  );
+
+  const frameworkPath = flavorToFrameworkPath(target, flavor);
+  const stat = await fs.lstat(path.join(frameworkPath, target));
+
+  // `_CodeSignature` is apparently generated only for simulator, afaik we don't need it.
+  await fs.remove(path.join(frameworkPath, '_CodeSignature'));
+
+  return {
+    target,
+    flavor,
+    frameworkPath,
+    binarySize: stat.size,
+  };
+}
+
+export async function cleanTemporaryFilesAsync(xcodeProject: XcodeProject) {
+  // Moves created xcframework to package folder
+  const xcFrameworkFilename = 'ExpoModulesCore.xcframework';
+  await fs.move(
+    path.join(OUT_OF_TREE_WORKING_DIR, xcFrameworkFilename),
+    path.join(
+      getExpoRepositoryRootDir(),
+      'packages',
+      'expo-modules-core',
+      'ios',
+      xcFrameworkFilename
+    )
+  );
+
+  // Cleanups working directory
+  await fs.remove(OUT_OF_TREE_WORKING_DIR);
+}
+
+async function createPodfileAsync(workDir: string) {
+  const content = `\
+platform :ios, '12.0'
+
+react_native_dir = File.dirname(\`node --print "require.resolve('react-native/package.json')"\`)
+require File.join(react_native_dir, "scripts/react_native_pods")
+require File.join(File.dirname(\`node --print "require.resolve('expo/package.json')"\`), "scripts/autolinking")
+
+target 'ExpoModulesCore_iOS' do
+  use_react_native!(
+    :path => react_native_dir
+  )
+end`;
+
+  await fs.writeFile(path.join(workDir, 'Podfile'), content);
+}
+
+async function createModulemapAsync(workDir: string) {
+  const content = `\
+framework module ExpoModulesCore {
+  umbrella header "ExpoModulesCore.h"
+
+  export *
+  module * { export * }
+}`;
+  await fs.writeFile(path.join(workDir, MODULEMAP_FILE), content);
+}
+
+async function createGeneratedHeaderAsync(workDir: string) {
+  const srcUmbrellaHeader = path.join(
+    getExpoRepositoryRootDir(),
+    'apps',
+    'bare-expo',
+    'ios',
+    'Pods',
+    'Target Support Files',
+    'ExpoModulesCore',
+    UMBRELLA_HEADER
+  );
+  assert(
+    await fs.pathExists(srcUmbrellaHeader),
+    `Cannot find ${UMBRELLA_HEADER}. Make sure to run \`et pods -f\` before prebuilding.`
```

Comment:
`pods` is just an alias to `pod-installs` — I'd prefer using the full name when we're referring to expotools commands (it's just more clear what the command does). Also, I guess the force flag is not required here 🤔

```suggestion
    `Cannot find ${UMBRELLA_HEADER}. Make sure to run \`et pod-install\` before prebuilding.`
```